pkg/clients/clients.go

```diff
@@ -375,6 +141,55 @@ func (dcs *DynamicClientSet) ResourceClientForObject(obj *unstructured.Unstructu
 	return dcs.ResourceClient(obj.GroupVersionKind(), obj.GetNamespace())
 }
 
+func (dcs *DynamicClientSet) gvkForKind(kind Kind) (gvk schema.GroupVersionKind, err error) {
+	resources, err := dcs.DiscoveryClientCached.ServerPreferredResources()
+	if err != nil {
+		return
+	}
+
+	for _, gvResources := range resources {
+		for _, resource := range gvResources.APIResources {
+			if resource.Kind == string(kind) {
+				gv := parseGVString(gvResources.GroupVersion)
+				gvk.Group, gvk.Version, gvk.Kind = gv.Group, gv.Version, resource.Kind
+				return
+			}
+		}
+	}
+
+	err = fmt.Errorf("failed to find gvk for Kind: %q", kind)
+	return
+}
+
+func (dcs *DynamicClientSet) namespaced(gvk schema.GroupVersionKind) bool {
+	resources, err := dcs.DiscoveryClientCached.ServerPreferredResources()
+	if err != nil {
+		// Default to true
+		return true
+	}
+
+	for _, gvResources := range resources {
+		if gvResources.GroupVersion == gvk.GroupVersion().String() {
+			for _, resource := range gvResources.APIResources {
+				if resource.Kind == gvk.Kind {
+					return resource.Namespaced
+				}
+			}
+		}
+	}
+
+	// Default to true
+	return true
+}
+
+func parseGVString(gv string) schema.GroupVersion {
```

Comment:
I feel like there should be an official parsing function for this?

Reply:
Ah, you're right. Will update.